sdk/eidolon_ai_sdk/agent/api_agent.py

```diff
@@ -0,0 +1,77 @@
+from typing import List, Optional, Any, cast
+
+import jsonref
+from pydantic import BaseModel, Field
+
+from eidolon_ai_client.util.logger import logger
+from eidolon_ai_sdk.agent.agent import register_action
+from eidolon_ai_sdk.builtins.logic_units.api_helper import build_call, get_content
+from eidolon_ai_sdk.builtins.logic_units.openapi_helper import build_actions, Action, Operation
+from eidolon_ai_sdk.system.reference_model import Specable
+from eidolon_ai_sdk.util.schema_to_model import schema_to_model
+
+
+class APIAgentSpec(BaseModel):
+    title: str = Field(description="Title of the API")
+    root_call_url: str = Field(description="Root URL of the API to call")
+    open_api_location: str = Field(description="Location of the OpenAPI schema")
+    operations_to_expose: List[Operation] = Field(description="Operations to expose")
+    extra_header_params: Optional[dict] = Field(description="Extra header parameters to add to every call. This can be a jinja template where the variables in the template are ENV variables (matching case)", default=None)
+    extra_query_params: Optional[dict] = Field(description="Extra query parameters to add to every call. This can be a jinja template where the variables in the template are ENV variables (matching case)", default=None)
```

Comment:
nit: better to keep the type as a dict and default to an empty dict. It will make to docs easier to read